test/core.jl

```diff
@@ -4684,9 +4684,15 @@ end
     f18725(x) = 2
 end
 @test Main.f18725(0) == 2
-@test_warn "WARNING: Method definition f18725(Any) in module Module18725" @eval Main module Module18725
-    f18725(x) = 1
-    f18725(x) = 2
+
+# PR #23030
+@test_nowarn @eval Main module Module23030
```

Comment:
right my mistake, I guess the previous version of this code that defines a replacement for Module18725 now shouldn't warn about the module overwriting itself, but should about overwriting the old module within Main? maybe?